### PR TITLE
Migrate voice command dialog

### PR DESCRIPTION
**Pull request by bramkragten**
And add text input to also support browsers that don't support speech to text.
image

image

### CODE REVIEW

File: src/panels/lovelace/hui-root.ts [hunk 1]
```diff
@@ -150,9 +150,11 @@ class HUIRoot extends LitElement {
                     .narrow=${this.narrow}
                   ></ha-menu-button>
                   <div main-title>${this.config.title || "Home Assistant"}</div>
-                  <ha-start-voice-button
-                    .hass="${this.hass}"
-                  ></ha-start-voice-button>
+                  <paper-icon-button
```
Comment (balloob) [inline src/panels/lovelace/hui-root.ts]: We should make sure it's only shown when the conversation integration is loaded.

File: src/panels/shopping-list/ha-panel-shopping-list.js [hunk 2]
```diff
@@ -72,10 +72,11 @@ class HaPanelShoppingList extends LocalizeMixin(PolymerElement) {
               narrow="[[narrow]]"
             ></ha-menu-button>
             <div main-title>[[localize('panel.shopping_list')]]</div>
-            <ha-start-voice-button
-              hass="[[hass]]"
-              can-listen="{{canListen}}"
-            ></ha-start-voice-button>
+            <paper-icon-button
```
Comment (balloob) [inline src/panels/shopping-list/ha-panel-shopping-list.js]: Same.

File: src/panels/states/ha-panel-states.js [hunk 3]
```diff
@@ -72,7 +72,11 @@ class PartialCards extends EventsMixin(NavigateMixin(PolymerElement)) {
             <div main-title="">
               [[computeTitle(views, defaultView, locationName)]]
             </div>
-            <ha-start-voice-button hass="[[hass]]"></ha-start-voice-button>
+            <paper-icon-button
```
Comment (balloob) [inline src/panels/states/ha-panel-states.js]: Same

File: src/translations/en.json [hunk 4]
```diff
@@ -554,6 +554,12 @@
       }
     },
     "dialogs": {
+      "voice_command": {
+        "did_not_hear": "<Home Assistant did not hear anything>",
```
Comment (balloob) [inline src/translations/en.json]: Let's move `<` and `>` out of the translations

**Review by balloob — APPROVED** [review]
Awesome! Ok to merge when comments addressed.

File: src/panels/lovelace/hui-root.ts [hunk 5]
```diff
@@ -405,6 +411,11 @@ class HUIRoot extends LitElement {
     ];
   }
 
+  protected firstUpdated(changedProperties: PropertyValues) {
+    super.updated(changedProperties);
+    this._conversation = isComponentLoaded(this.hass, "conversation");
```
Comment (balloob) [inline src/panels/lovelace/hui-root.ts]: This is not correct. We should check it on every render. If a user installs Almond, we want this icon to show up. You could memoize it though on `hass.config.components`.

Comment (bramkragten) [reply src/panels/lovelace/hui-root.ts]: Yeah of course 🤦‍♂

Comment (balloob) [reply src/panels/lovelace/hui-root.ts]: We might want to create a new `isComponentLoaded` that takes hass.config.components and so is better to memoize.

File: src/panels/lovelace/hui-root.ts [hunk 6]
```diff
@@ -60,10 +61,13 @@ class HUIRoot extends LitElement {
   @property() private _routeData?: { view: string };
   @property() private _curView?: number | "hass-unused-entities";
   private _viewCache?: { [viewId: string]: HUIView };
-  @property() private _conversation = false;
 
   private _debouncedConfigChanged: () => void;
 
+  private _conversation = memoizeOne((_components) =>
+    isComponentLoaded(this.hass, "conversation")
```
Comment (balloob) [inline src/panels/lovelace/hui-root.ts]: Hacky but Smart.

File: src/panels/lovelace/hui-root.ts (hunk 5 above)
Comment (bramkragten) [reply src/panels/lovelace/hui-root.ts]: Do we need to optimise it for states and shopping list or is checking it on every change of hass ok?

Comment (balloob) [reply src/panels/lovelace/hui-root.ts]: I just care about Lovelace right now.

**Comment by balloob** [comment]
Going to merge this so I can use it in my demo :)

**Pull request merged**